app/controllers/search_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SearchController < ApplicationController
+  def new
+    @search = Search.new
+  end
+
+  def create
+    search = Search.new(params[:search])
+
+    @results = search.results
+
+    render :results
+  end
+
+  def search
+    @search = if params[:search]
```

Comment:
Can this be simplified to just Search.new(params[:search]) allways? If the function handles nil as an argument.